File: 2017/04/high-entropy.go
package main

import "fmt"
import "os"
import "io/ioutil"
import "strings"


func isValid(passphrase string) bool {
    words := strings.Split(passphrase, " ")
    seen := make(map[string]bool)
    for _, word := range words {
        if seen[word] == true {
            return false
        }
        seen[word] = true
    }
    return true
}


func main() {
    data, err := ioutil.ReadFile(os.Args[1])
    if err != nil {
        panic(err)
    }
    contents := strings.Trim(string(data), "\n")
    passphrases := strings.Split(contents, "\n")
    numValid := 0
    for _, passphrase := range passphrases {
        if isValid(passphrase) {
            numValid += 1
        }
    }

    fmt.Println(numValid)
}

File: 2017/04/high-entropy.py
# -*- coding: utf-8 -*-
import sys


def is_valid(passphrase):
    words = passphrase.split()
    return len(set(words)) == len(words)


def main(filename):
    with open(filename) as f:
        passphrases = f.read().strip().splitlines()

    print(sum(is_valid(p) for p in passphrases))


if __name__ == '__main__':
    main(sys.argv[1])
